nototools/generate_coverage_data.py

```diff
@@ -0,0 +1,253 @@
+    indent += '  '
+    sfx = '\n' + indent
+    root.text = sfx
+    for elem in root:
+      elem.tail = sfx
+      _prettify(elem, indent)
+    elem.tail = sfx[:-2]
+
+
+def _build_tree(coveragedata):
+  root = ET.Element('coveragedata')
+  root.append(_build_meta_elem(coveragedata.metadata))
+  root.append(_build_cmap_elem(coveragedata.cmapdata))
+  _prettify(root)
+  return ET.ElementTree(element=root)
+
+
+def write(coveragedata, out_file=None):
+  """Write coverage data to xml."""
+  tree = _build_tree(coveragedata)
+  if out_file:
+    tree.write(out_file, encoding='utf-8', xml_declaration=True)
+  else:
+    print ET.tostring(tree.getroot(), encoding='utf-8')
+
+
+def _read_meta(meta_elem):
+  date = meta_elem.get('date')
+  program = meta_elem.get('program')
+  args = []
+  args_node = meta_elem.find('args')
+  if args_node is not None:
+    for arg_node in args_node:
+      key = arg_node.tag
+      if arg_node.attrib.get('isList', False):
+        prefix = arg_node.attrib.get('prefix', '')
+        val = []
+        for item in arg_node.findall('item'):
+          val.append(prefix + item.text)
+      else:
+        val = arg_node.get('val').strip()
+      args.append((key, val))
+  return MetaData(date, program, args)
+
+
+def _read_data(data_elem):
+  name = data_elem.find('name').text.strip()
+  ranges = data_elem.find('ranges').text.strip()
+  return CmapData(name, ranges)
+
+
+def _read_tree(root_elem):
+  meta = _read_meta(root_elem.find('meta'))
+  data = _read_data(root_elem.find('data'))
+  return CoverageData(meta, data)
+
+
+def read(coveragefile):
+  return _read_tree(ET.parse(coveragefile).getroot())
+
+
+def main():
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '-o', '--output_file', help='name of xml file to output', metavar='file')
+  parser.add_argument(
+      '-d', '--dirs', help='directories containing font files', metavar='dir',
+      nargs='+')
+  parser.add_argument(
+      '-f', '--files', help='font files', metavar='file', nargs='+')
+  parser.add_argument(
+      '-n', '--name', help='short name of this collection, used in reports',
+      metavar='name', required=True)
+  parser.add_argument(
+      '-c', '--cmap_data', help='cmap data file (default %s)' %
+      default_coverage_file, const=default_coverage_file, nargs='?',
+      metavar='file')
+  args = parser.parse_args()
+
+  if args.dirs or args.files:
+    paths = tool_utils.collect_paths(args.dirs, args.files)
+    cps, paths = get_cps_from_files(paths)
+  elif args.cmap_data:
+    cps = get_cps_from_cmap_data_file(args.cmap_data)
+    paths = None
+  else:
+    print 'Please specify font files, directories, or a cmap data file.'
+    return
+  coverage = create(args.name, cps, paths=paths, cmap_data=args.cmap_data)
+  write(coverage, args.output_file)
+
+  # test
+  if args.output_file:
+    new_coverage = read(args.output_file)
+    if new_coverage == coverage:
+      print 'OK'
+    else:
+      print 'FAIL'
```

Comment:
Did you mean to keep this test in? If so, suggesting to make this message more informative.